handlers/config.go

```diff
@@ -183,10 +185,14 @@ func getPrometheusConfig() PrometheusConfig {
 
 		flags, err := client.GetFlags()
 		if checkErr(err, "Failed to fetch Prometheus flags") {
-			if retentionString, ok := flags["storage.tsdb.retention"]; ok {
+			if retentionString, ok := flags["storage.tsdb.retention.time"]; ok {
 				retention, err := model.ParseDuration(retentionString)
-				if checkErr(err, fmt.Sprintf("Invalid storage.tsdb.retention [%s]", retentionString)) {
-					promConfig.StorageTsdbRetention = int64(time.Duration(retention).Seconds())
+				if checkErr(err, fmt.Sprintf("Invalid storage.tsdb.retention.time [%s]", retentionString)) {
+					if retention == 0 {
+						log.Warningf("%s", "Prometheus storage.tsdb.retention.time configured to 0, ignoring...")
```

Comment:
just curious - why is this one big "%s" ?? :) Why not just:

```suggestion
						log.Warning("Prometheus storage.tsdb.retention.time configured to 0, ignoring...")
```